app/routes.py

```diff
@@ -1,2 +1,274 @@
-from flask import Blueprint
+import requests
+from dotenv import load_dotenv
+import os
+import datetime
+from sqlalchemy import desc
+from app import db
+from app.models.task import Task
+from app.models.goal import Goal
+from flask import request, Blueprint, make_response, jsonify
+tasks_bp = Blueprint("tasks", __name__, url_prefix="/tasks")
+goals_bp = Blueprint("goals", __name__, url_prefix="/goals")
 
+load_dotenv()
+
+
+@tasks_bp.route("/", methods=["GET", "POST"])
+def handle_tasks():
+    if request.method == "GET":
+        sort_query = request.args.get("sort")
+        if sort_query == "asc":
+            tasks = Task.query.order_by(Task.title)
+        elif sort_query == "desc":
+            tasks = Task.query.order_by(desc(Task.title))
+        else:
+            tasks = Task.query.all()
+
+        tasks_response = []
+        for task in tasks:
+            if not task.completed_at:
+                is_complete = False
+            else:
+                is_complete = True
+            tasks_response.append({
+                "id": task.task_id,
+                "title": task.title,
+                "description": task.description,
+                "is_complete": is_complete
+            })
+        return make_response(jsonify(tasks_response), 200)
+
+    elif request.method == "POST":
+        request_body = request.get_json()
+        keys = ['title', 'description', 'completed_at']
+        for key in keys:
+            if key not in request_body:
+                task_response = {"details": "Invalid data"}
+                return make_response(task_response, 400)
+
+        new_task = Task(title=request_body["title"],
+                        description=request_body["description"],
+                        completed_at=request_body["completed_at"])
+
+        db.session.add(new_task)
+        db.session.commit()
+        if not request_body["completed_at"]:
+            is_complete = False
+        else:
+            is_complete = True
+        task_response = {"task":
+                         {"id": new_task.task_id,
+                          "title": new_task.title,
+                          "description": new_task.description,
+                          "is_complete": is_complete}
+                         }
+        return make_response(task_response, 201)
+
+
+@tasks_bp.route("/<task_id>", methods=["GET", "PUT", "DELETE"])
+def handle_task(task_id):
+    task = Task.query.get(task_id)
+    if not task:
+        return make_response(f"Task #{task_id} Not Found", 404)
```

Comment:
👍 

We could also consider using `get_or_404`.

We _do_ need to do this check for GET, PUT, and DELETE requests, but we could still think about splitting these into separate functions (e.g. `get_task`, `update_task`, and `delete_task`).